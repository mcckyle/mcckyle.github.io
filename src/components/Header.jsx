//Filename: Header.jsx
//Author: Kyle McColgan
//Date: 28 August 2025
//Description: Contains the React component for the website contact header section.

import React from 'react';
import profilePic from '/images/kdm-ocean-01.png';

function Header()
{
    return (
        <header className="site-header">
            <div className="header-text">
                <h1>Kyle D. McColgan</h1>
                <h2>Technology Professional</h2>
            </div>
            <img className="header-img" src={profilePic} alt="Kyle at the beach." />
        </header>
    );
}

export default Header;
