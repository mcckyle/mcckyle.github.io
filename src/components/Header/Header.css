/* Filename: Header.css */

/* ====================================
   Header Container Styles.
======================================= */
.site-header
{
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;

    width: 100%;
    padding: clamp(4rem, 8vw, 7rem) clamp(1.5rem, 6vw, 3rem);
    overflow: hidden;

    background: linear-gradient(
      to bottom,
      color-mix(in srgb, var(--surface) 95%, var(--primary) 5%) 0%,
      color-mix(in srgb, var(--surface-alt) 95%, transparent 5%) 100%
    );

    border-bottom: 1px solid rgba(255, 255, 255, 0.05)
}

.site-header::before
{
    content: "";
    position: absolute;
    inset: 0;
    background: radial-gradient(
      circle at 50% 35%,
      rgba(255, 255, 255, 0.04) 0%,
      transparent 75%
    );

    pointer-events: none;
    z-index: 0;
}

/* ====================================
   Header Content Styles.
======================================= */
.header-content
{
    position: relative;
    z-index: 1;

    display: flex;
    flex-direction: column;
    align-items: center;
    gap: clamp(2rem, 4vw, 3rem);
    max-width: 960px;
    width: 100%;
    margin: 0 auto;
}

/* =========================================
   Header Image Presentation Styles.
============================================ */
.header-image-wrapper
{
    max-width: 380px;
    width: 100%;
    aspect-ratio: 1 / 1;
}

.header-img
{
    width: 100%;
    height: 100%;
    border-radius: var(--radius-lg);
    object-fit: cover;
    filter: brightness(0.98) saturate(1.05);
    box-shadow: 0 8px 26px rgba(0, 0, 0, 0.15);

    transition: transform var(--transition),
                filter var(--transition),
                box-shadow var(--transition);
}

.header-img:hover
{
    transform: scale(1.015);
    filter: brightness(1.02) saturate(1.12);
    box-shadow: 0 14px 34px rgba(0, 0, 0, 0.18);
}

/* ====================================
   Header Text Styles.
======================================= */
.header-text
{
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 0.75rem;
}

.header-name
{
    font-family: "Playfair Display", serif;
    font-size: clamp(2.75rem, 6vw, 3.75rem);
    font-weight: 700;
    color: var(--text-strong);
    margin: 0;

    letter-spacing: 1px;
    line-height: 1.05;
    text-shadow: 0 2px 3px rgba(0, 0, 0, 0.05);
}

.header-name::after
{
    content: "";
    display: block;
    height: 3px;
    width: 80px;
    margin: 1rem auto 0;
    border-radius: 999px;

    background: linear-gradient(90deg, var(--primary), var(--accent));
    opacity: 0.9;
}

.header-role
{
    font-size: clamp(1rem, 1.8vw, 1.25rem);
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 2px;
    color: var(--accent);
    opacity: 0.9;
}

/* ====================================
   Accessibility Utilities Styles.
======================================= */
.visually-hidden
{
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    white-space: nowrap;
    border: 0;
}
