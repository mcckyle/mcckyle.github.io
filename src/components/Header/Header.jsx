//Filename: Header.jsx
//Author: Kyle McColgan
//Date: 26 October 2025
//Description: This file contains the component for the personal React project header section.

import React from "react";
import profilePic from "/images/kdm-ocean-01.png";

import "./Header.css";

function Header()
{
    return (
      <header className="site-header" role="banner">
        <div className="header-content">
          <figure className="header-image-wrapper">
            <img
              className="header-img"
              src={profilePic}
              alt="Kyle on a boat by the beach."
              loading="lazy"
              decoding="async"
            />
            <figcaption className="visually-hidden">
              Portrait of Kyle D. McColgan on a boat by the beach.
            </figcaption>
          </figure>

          <div className="header-text">
            <h1 className="header-name">Kyle D. McColgan</h1>
            <p className="header-role">Technology Professional</p>
          </div>
        </div>
      </header>
    );
}

export default Header;
