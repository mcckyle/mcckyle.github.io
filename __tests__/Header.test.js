//Filename: Header.test.js
//Author: Kyle McColgan
//Date: 28 August 2025
//Description: This file contains the Jest unit tests for my personal website header section.

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../src/components/Header';

it('renders the header with the correct title', () => {
    render(<Header />);
    const title = screen.getByRole('heading', { level: 1 });
    expect(title).toHaveTextContent('Kyle D. McColgan');
});

it('renders the header with the correct subtitle', () => {
    render(<Header />);
    const subtitle = screen.getByRole('heading', { level: 2 });
    expect(subtitle).toHaveTextContent('Technology Professional');
});

it('renders the header image', () => {
    render(<Header />);
    const image = screen.getByAltText('Kyle at the beach.');
    expect(image).toBeInTheDocument();
});

it('has the correct image source', () => {
    render(<Header />);
    const image = screen.getByAltText('Kyle at the beach.');
    expect(image).toHaveAttribute('src', 'test-file-stub');
});

it('has the correct alt text for the image', () => {
    render(<Header />);
    const image = screen.getByAltText('Kyle at the beach.');
    expect(image).toHaveAttribute('alt', 'Kyle at the beach.');
});

it('renders the image with the correct class', () => {
    render(<Header />);
    const image = screen.getByAltText('Kyle at the beach.');
    expect(image).toHaveClass('header-img');
});

it('renders the header element', () => {
    render(<Header />);
    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();
});

it('does not log any errors to the console', () => {
    const consoleError = jest.spyOn(console, 'error').mockImplementation();
    render(<Header />);
    expect(consoleError).not.toHaveBeenCalled();
});

it('ensures the header is not empty', () => {
    render(<Header />);
    const header = screen.getByRole('banner');
    expect(header).not.toBeEmptyDOMElement();
});

it('matches the snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
});
