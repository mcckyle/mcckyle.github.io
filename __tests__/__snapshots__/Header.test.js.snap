// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="site-header"
    role="banner"
  >
    <div
      class="header-content"
    >
      <figure
        class="header-image-wrapper"
      >
        <img
          alt="Kyle on a boat by the beach."
          class="header-img"
          decoding="async"
          loading="lazy"
          src="test-file-stub"
        />
        <figcaption
          class="visually-hidden"
        >
          Portrait of Kyle D. McColgan on a boat by the beach.
        </figcaption>
      </figure>
      <div
        class="header-text"
      >
        <h1
          class="header-name"
        >
          Kyle D. McColgan
        </h1>
        <p
          class="header-role"
        >
          Technology Professional
        </p>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
